staging/src/k8s.io/kubectl/pkg/cmd/describe/describe.go

```diff
@@ -69,38 +69,81 @@ var (
 		kubectl describe pods frontend`))
 )
 
-type DescribeOptions struct {
-	CmdParent string
-	Selector  string
-	Namespace string
-
-	Describer  func(*meta.RESTMapping) (describe.ResourceDescriber, error)
-	NewBuilder func() *resource.Builder
-
-	BuilderArgs []string
-
-	EnforceNamespace bool
-	AllNamespaces    bool
-
-	DescriberSettings *describe.DescriberSettings
+// DescribeFlags directly reflect the information that CLI is gathering via flags. They will be converted to Options,
+// which reflect the runtime requirements for the command.
+type DescribeFlags struct {
+	Factory           cmdutil.Factory
```

Comment:
This is wrong, `DescribeFlags` should ONLY contain flags exposed by the command, `Factory`'s place is only at `DescribeOptions` struct.

Reply:
Also you should be easily replace the `Factory` with `ResourceBuilderFlags` and `RESTClientGetter`, see how https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go is structured, that's what I'd want to see applied across the board.

Reply:
Yeah thanks for addressing this, I'll create a follow up pr on this, but I guess we'll also have to refactor some commands for which changes are already done and merged. For example :- [apply command](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go#L55) also uses `Factory`

Reply:
Thanks @PuneetPunamiya for noticing that

@soltysh I removed the Factory in the struct in this PR https://github.com/kubernetes/kubernetes/pull/114030